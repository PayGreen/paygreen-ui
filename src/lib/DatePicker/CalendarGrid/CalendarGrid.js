import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import CalendarGridBase from './style/index';
import CalendarCell from '../CalendarCell/CalendarCell';

/**
 * @desc Return the first and last days of Month using month index.
 * @param {number} month
 */
const getMonthBoundaries = month => {
    const startOfMonth = moment()
        .month(month)
        .startOf('M');

    const endOfMonth = moment()
        .month(month)
        .add(1, 'M')
        .startOf('M');

    return [startOfMonth, endOfMonth];
};

const CalendarGrid = ({ month, ...rest }) => {
    const [startOfMonth, endOfMonth] = getMonthBoundaries(month);

    const daysOfMonth = [];
    let daysCursor = moment(startOfMonth);
    do {
        daysOfMonth.push(
            <CalendarCell
                key={daysCursor.toDate()}
                date={daysCursor.clone()}
                {...rest}
            />,
        );
    } while (daysCursor.add(1, 'days').diff(endOfMonth) < 0);

    return (
        <CalendarGridBase offset={startOfMonth.isoWeekday()} {...rest}>
            {daysOfMonth}
        </CalendarGridBase>
    );
};

CalendarGrid.propTypes = {
    month: PropTypes.number,
};

CalendarGrid.defaultProps = {
    month: moment().month(),
};

export default CalendarGrid;
