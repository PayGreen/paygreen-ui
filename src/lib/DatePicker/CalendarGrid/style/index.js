import styled from 'styled-components';

const CalendarGridBase = styled.div`
    display: grid;

    grid-template-columns: repeat(7, auto);
    grid-gap: ${props => props.theme.line};
    justify-content: start;
    
    width: calc(
        (${props => props.theme.icon.size.lg} * 7) +
            (${props => props.theme.line} * 8)
    );
    height: auto;
    padding: ${props => props.theme.line};

    border-bottom-left-radius: ${props => props.theme.radius.lg};
    border-bottom-right-radius: ${props => props.theme.radius.lg};

    background-color: ${props => props.theme.wab.white00};

    div:first-child {
        grid-column: ${props => props.offset};
    }
`;

export default CalendarGridBase;
