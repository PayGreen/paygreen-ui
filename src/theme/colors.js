const mainColors = {
    green: '#33ad73',
    blue: '#00a99d',
    orange: '#e58a28',
    pink: '#ec6a6a',
    purple: '#a26ae1',
    red: '#b54025',
};

const colors = {
    green: {
        main: mainColors.green,
        gradientBase: '#77d1a6',
        gradientShade: '#a4db95',
    },
    blue: {
        main: mainColors.blue,
        gradientBase: '#51c6be',
        gradientShade: '#76bbd8',
    },
    orange: {
        main: mainColors.orange,
        gradientBase: '#fab56b',
        gradientShade: '#faa272',
    },
    pink: {
        main: mainColors.pink,
        gradientBase: '#e98e90',
        gradientShade: '#e98eb9',
    },
    purple: {
        main: mainColors.purple,
        gradientBase: '#b887ee',
        gradientShade: '#dd94ef',
    },
};

export { mainColors, colors };
