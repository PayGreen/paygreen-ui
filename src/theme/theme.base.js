const responsive = {
    sm: 550,
    md: 960,
    lg: 1200,
    xl: 1360,
};

const query = (value, operator) => {
    if (operator == 'max') {
        value -= 1;
    }
    return operator + '-width: ' + value + 'px';
};

const pallet = {
    green: {
        main: '#33ad73',
        light: '#d5f3e5',
    },
    blue: {
        main: '#00a99d',
        light: '#e1f3f2',
    },
    orange: {
        main: '#e58a28',
        light: '#fbe6ce',
    },
};

export const ThemeBase = {
    name: 'Base',
    color: {
        primary: {
            ...pallet.green,
            gradientBase: '#77d1a6',
            gradientShade: '#a4db95',
        },
        secondary: {
            ...pallet.blue,
            gradientBase: '#51c6be',
            gradientShade: '#76bbd8',
        },
        tertiary: {
            ...pallet.orange,
            gradientBase: '#fab56b',
            gradientShade: '#faa272',
        },
        quaternary: {
            main: '#ec6a6a',
            light: '#ffe0e1',
            gradientBase: '#e98e90',
            gradientShade: '#e98eb9',
        },
        quinary: {
            main: '#a26ae1',
            light: '#e9d9fb',
            gradientBase: '#b887ee',
            gradientShade: '#dd94ef',
        },
    },
    status: {
        success: {
            ...pallet.green,
        },
        default: {
            ...pallet.blue,
        },
        warning: {
            ...pallet.orange,
        },
        danger: {
            main: '#b54025',
            light: '#f7d7d0',
        },
    },
    font: {
        size: {
            tiny: '1rem',
            xxs: '1.2rem',
            xs: '1.4rem',
            sm: '1.6rem',
            base: '1.8rem',
            md: '2.1rem',
            lg: '3rem',
            xl: '4rem',
        },
        weight: {
            normal: 400,
            bold: 700,
        },
        lineHeight: {
            base: 1,
            md: 1.5,
        },
        spacing: '0.3em',
        underline: {
            space: {
                tiny: '8px',
                xxs: '10px',
                xs: '10px',
                sm: '12px',
                base: '12px',
                md: '12px',
                lg: '24px',
                xl: '24px',
            },
            line: {
                tiny: '2px',
                xxs: '2px',
                xs: '2px',
                sm: '2px',
                base: '2px',
                md: '2px',
                lg: '3px',
                xl: '3px',
            },
            width: {
                tiny: '30px',
                xxs: '40px',
                xs: '40px',
                sm: '50px',
                base: '50px',
                md: '65px',
                lg: '80px',
                xl: '80px',
            },
        },
    },
    line: '2px',
    dash: '3px',
    radius: {
        none: '0',
        sm: '4px',
        md: '8px',
        lg: '12px',
    },
    arrow: {
        sm: '8px',
        md: '12px',
        lg: '16px',
    },
    space: {
        none: '0',
        xs: '4px',
        sm: '8px',
        md: '16px',
        lg: '32px',
        xl: '64px',
    },
    grid: {
        header: '45px',
        headerBig: '70px',
        topbar: '45px',
        sidebar: '380px',
        maxWidth: '1400px',
    },
    blockShift: {
        none: '0',
        xs: '64px',
        sm: '128px',
        md: '256px',
        lg: '320px',
        xl: '384px',
    },
    blockSpace: {
        none: '0',
        xs: '16px',
        sm: '24px',
        md: '40px',
        lg: '52px',
        xl: '68px',
    },
    blockWidth: {
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '800px',
    },
    blockHeader: {
        sm: '140px',
        md: '180px',
        lg: '210px',
        xl: '240px',
    },
    bannerHeight: {
        xs: '33vh',
        sm: '50vh',
        lg: '100vh',
    },
    wave: {
        simple: '8vw',
        double: '11vw',
    },
    icon: {
        size: {
            xxs: '12px',
            xs: '16px',
            sm: '20px',
            md: '24px',
            lg: '32px',
            xl: '40px',
        },
        shift: {
            xxs: '2px',
            xs: '3px',
            sm: '4px',
            md: '4px',
            lg: '6px',
            xl: '6px',
        },
    },
    imageSize: {
        auto: 'auto',
        xxs: '150px',
        xs: '240px',
        sm: '400px',
        md: '560px',
        lg: '640px',
        xl: '760px',
    },
    imageCircleSize: {
        auto: 'auto',
        xxs: '38px',
        xs: '60px',
        sm: '80px',
        md: '100px',
        lg: '120px',
        xl: '140px',
    },
    logoHeight: {
        none: 'auto',
        xs: '20px',
        sm: '30px',
        md: '40px',
        lg: '50px',
        xl: '60px',
    },
    logoWidth: {
        none: 'auto',
        xs: '90px',
        sm: '120px',
        md: '150px',
        lg: '200px',
        xl: '250px',
    },
    loaderSize: {
        xxs: '40px',
        xs: '50px',
        sm: '60px',
        md: '70px',
        lg: '80px',
        xl: '90px',
    },
    transition: {
        xxs: '0.1s',
        xs: '0.2s',
        sm: '0.3s',
        md: '0.5s',
        lg: '0.7s',
        xl: '1s',
    },
    zindex: {
        hidden: -100,
        layer: -5,
        background: -1,
        base: 0,
        content: 1,
        corner: 3,
        front: 5,
        loader: 90,
        subHeader: 100,
        menu: 105,
        header: 110,
        popin: 115,
        modal: 120,
    },
    query: {
        min: {
            sm: query(responsive.sm, 'min'),
            md: query(responsive.md, 'min'),
            lg: query(responsive.lg, 'min'),
            xl: query(responsive.xl, 'min'),
        },
        max: {
            sm: query(responsive.sm, 'max'),
            md: query(responsive.md, 'max'),
            lg: query(responsive.lg, 'max'),
            xl: query(responsive.xl, 'max'),
        },
    },
    button: {
        paddingWidth: {
            sm: '22px',
            md: '26px',
            lg: '30px',
        },
        paddingHeight: {
            sm: '12px',
            md: '14px',
            lg: '16px',
        },
        shift: '3px',
        font: {
            sm: '1.2rem',
            md: '1.4rem',
            lg: '1.5rem',
        },
        letterSpacing: {
            sm: '0.3rem',
            md: '0.4rem',
            lg: '0.45rem',
        },
    },
    daButton: {
        buttonHeight: {
            sm: '32px',
            md: '38px',
            lg: '44px',
        },
        shift: {
            sm: '2px',
            md: '4px',
            lg: '6px',
        },
        font: {
            sm: '1.35rem',
            md: '1.4rem',
            lg: '1.7rem',
        },
        iconSpacing: {
            sm: '0.9rem',
            md: '1rem',
            lg: '1.1rem',
        },
    },
    form: {
        inputWidth: {
            xs: '260px',
            sm: '330px',
            md: '400px',
            lg: '500px',
        },
        textareaHeight: '200px',
        radio: {
            md: '120px',
            lg: '150px',
        },
        checkboxSize: {
            sm: '18px',
            md: '20px',
            lg: '22px',
        },
        radioButtonSize: {
            sm: '19px',
            md: '21px',
            lg: '23px',
        },
        radioButtonSpace: {
            sm: '8px',
            md: '10px',
            lg: '10px',
        },
        toggle: '30px',
    },
    corner: {
        size: {
            lg: '34px',
        },
        padding: {
            lg: '6px',
        },
    },
    shadow: {
        size: {
            none: '0 0 0',
            xs: '1px 1px 5px',
            sm: '1px 1px 10px',
            md: '2px 2px 15px',
            lg: '3px 3px 20px',
        },
        opacity: {
            none: 1,
            xs: 0.95,
            sm: 0.9,
            md: 0.8,
            lg: 0.7,
        },
    },
    rotateSize: {
        d0: '0deg',
        d45: '45deg',
        d90: '90deg',
        d135: '135deg',
        d180: '180deg',
        d225: '225deg',
        d270: '270deg',
        d315: '315deg',
    },
    dataBar: {
        height: '10px',
        width: {
            sm: '300px',
            md: '400px',
            lg: '500px',
            xl: '600px',
        },
    },
};
