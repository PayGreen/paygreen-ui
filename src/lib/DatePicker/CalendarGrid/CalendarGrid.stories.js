import React, { useState } from 'react';
import moment from 'moment';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import { folder } from '../../../shared/constants';
import CalendarGrid from './CalendarGrid';
import { MonthContextProvider } from '../context/MonthContext';
import { DateContextProvider } from '../context/DateContext';

storiesOf(folder.form + folder.sub.datePicker + 'CalendarGrid', module)
    .addDecorator(withKnobs)
    .add('CalendarGrid', () => {
        const [selectedDate, setSelectedDate] = useState(moment().startOf('D'));
        const [month, setMonth] = useState(moment().month());

        return (
            <DateContextProvider value={[selectedDate, setSelectedDate]}>
                <MonthContextProvider value={[month, setMonth]}>
                    <CalendarGrid />
                </MonthContextProvider>
            </DateContextProvider>
        );
    });
