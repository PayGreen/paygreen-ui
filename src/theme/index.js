import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
