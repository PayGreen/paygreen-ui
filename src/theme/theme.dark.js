import patternDark from './assets/pattern-dark.png';
import { baseDark, gradients } from './colors';
import { ThemeBase } from './theme.base';

const ThemeDark = {
    ...ThemeBase,
    name: 'Dark',
    color: {
        primary: {
            ...baseDark.green,
            ...gradients.green,
        },
        secondary: {
            ...baseDark.blue,
            ...gradients.blue,
        },
        tertiary: {
            ...baseDark.orange,
            ...gradients.orange,
        },
        quaternary: {
            ...baseDark.pink,
            ...gradients.pink,
        },
        quinary: {
            ...baseDark.purple,
            ...gradients.purple,
        },
    },
    status: {
        success: {
            ...baseDark.green,
        },
        default: {
            ...baseDark.blue,
        },
        warning: {
            ...baseDark.orange,
        },
        danger: {
            ...baseDark.red,
        },
    },
    wab: {
        white00: '#202020',
        white10: '#2c2c2c',
        white20: '#383838',

        grey10: '#494949',
        grey20: '#5b5b5b',
        grey30: '#7f7f7f',
        grey40: '#a2a2a2',
        grey50: '#cfcfcf',
        grey60: '#eeeeee',

        black00: '#ffffff',
    },
    pattern: patternDark,
    logoColor: '#ffffff',
};

export { ThemeDark };
