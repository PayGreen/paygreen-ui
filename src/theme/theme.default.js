import { ThemeBase } from './theme.base';
import patternLight from './assets/pattern-light.png';

let ThemeDefault = { ...ThemeBase };
ThemeDefault = Object.assign(ThemeDefault, {
    name: 'Default',
    wab: {
        white00: '#ffffff',
        white10: '#fafafa',
        white20: '#f6f6f6',

        grey10: '#eeeeee',
        grey20: '#dbdbdb',
        grey30: '#aaaaaa',
        grey40: '#858585',
        grey50: '#636363',
        grey60: '#434343',

        black00: '#000000',
    },
    pattern: patternLight,
});

export { ThemeDefault };
