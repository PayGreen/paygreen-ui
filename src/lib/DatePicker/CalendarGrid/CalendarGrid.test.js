import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../../theme';
import CalendarGrid from './CalendarGrid';

it('renders without crashing', () => {
    const calendarGrid = TestRenderer.create(
        <CalendarGrid theme={ThemeDefault} />,
    );
    expect(calendarGrid.toJSON()).toMatchSnapshot();
});
