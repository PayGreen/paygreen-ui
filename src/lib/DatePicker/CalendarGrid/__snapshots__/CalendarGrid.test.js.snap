// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="style__CalendarGridBase-sc-1qui6a5-0 flnYys"
>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    3
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    4
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 bfGwSB"
    onClick={[Function]}
  >
    5
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    6
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    7
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    8
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    9
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    10
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    11
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    12
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    13
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    14
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    15
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    16
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    19
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    20
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    21
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    22
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    23
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    24
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    25
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    26
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    27
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    28
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    29
  </button>
  <button
    className="style__CalendarCellBase-sc-17uxo6m-0 jMhBbg"
    onClick={[Function]}
  >
    30
  </button>
</div>
`;
